src/Microsoft.Azure.WebJobs.Host/Singleton/IDistributedLockManager.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Manage distributed locks. A lock is specified by (account, lockId). 
+    /// </summary>
+    /// <remarks>
+    /// The default implementation of this is based on blob leases. 
+    /// 1. Account can be null or it may be a storage account name intended for <see cref="IStorageAccountProvider"/>. 
+    /// 2. lockId has the same naming restrictions as blobs. 
+    /// 3. For distributed locks, this implementation is responsible for renewing the locks and setting <see cref="IDistributedLock.LeaseLost"/> if the lock can't be renewed. 
```

Comment:
Latest iteration does this. 

---
In reply to: [120951100](https://github.com/Azure/azure-webjobs-sdk/pull/1184#discussion_r120951100) [](ancestors = 120951100)